k2/csrc/determinize.h

```diff
@@ -766,8 +756,10 @@ class DetStateMap {
                          const Fsa &fsa, std::pair<uint64_t, uint64_t> *vec) {
     assert(d.normalized);
 
-    uint64_t a = d.state_id + 17489 * d.seq_len,
-             b = d.state_id * 103979 + d.seq_len;
+    int32_t input_state_id = d.elements.begin()->first;
+
+    uint64_t a = input_state_id + 17489 * d.seq_len,
```

Comment:
Remove input_state_id here, just have 17489 * d.seq_len and so on... and I'll say below..

Reply:
well, around line 776, insert: 
```
    a += elem->state_id;   // This is `base_state`: the state from which we 
                                          // start (and accept the specified symbol sequence).
```
(we don't need it in b, actually).

Reply:
actually, no, to reduce the chance of collisions, do instead:
       a = elem->state_id + 14051 * a

Reply:
.. I mean collisions on a, which is used for hashing.  collisions should still be vanishingly rare if we take a and b together.

Reply:
oh, add the `base_state`, got it, thanks!

Reply:
Updated